doctr/models/recognition/core.py

```diff
@@ -58,16 +58,11 @@ def resize(
         Returns:
             the processed image after being resized
         """
-        image_shape = tf.shape(x)
-        image_height = tf.cast(image_shape[0], dtype=tf.float32)
-        image_width = tf.cast(image_shape[1], dtype=tf.float32)
 
-        scale = self.output_size[0] / image_height
-        max_width = tf.cast(self.output_size[1], tf.int32)
-        new_width = tf.minimum(tf.cast(scale * image_width, dtype=tf.int32), max_width)
-
-        resized = tf.image.resize(x, [self.output_size[0], new_width], method=self.interpolation)
-        padded = tf.image.pad_to_bounding_box(resized, 0, 0, self.output_size[0], self.output_size[1])
+        # Preserve aspect ratio during resizing
+        resized = tf.image.resize(x, self.output_size, method=self.interpolation, preserve_aspect_ratio=True)
```

Comment:
How can you be sure that your resizing will be under max_width and max_height if you preserve aspect ratio ?

Reply:
Sorry, it's written in the doc : https://www.tensorflow.org/api_docs/python/tf/image/resize
and I tried on my hand and can confirm this is working